modepy/shapes.py

```diff
@@ -264,7 +264,8 @@ def face_normal(face: Face, normalize: bool = True) -> np.ndarray:
     volume_vertices = unit_vertices_for_shape(face.volume_shape)
     face_vertices = volume_vertices[:, face.volume_vertex_indices]
 
-    if face.dim == 0:
+    dim = getattr(face, "dim", face.volume_shape.dim - 1)
```

Comment:
This is a bit unfortunate, but couldn't figure out how to convince mypy that `Face` is a mixin and all subclasses are expected to have a `dim` (?).